dateparser/freshness_date_parser.py

```diff
@@ -91,7 +94,7 @@ def apply_time(dateobj, timeobj):
         date, period = self._parse_date(date_string, settings.PREFER_DATES_FROM)
 
         if date:
-            date = apply_time(date, _time)
+            date, period = apply_time(date, _time, period)
```

Comment:
:lipstick: I personally think that checking if the return value is the same as the passed `date` object after this line, and changing `period` based on that, would look better than the proposed changes to the implementation of `apply_time`.

Reply:
Made this change.